src/main/java/mekanism/common/integration/multipart/MultipartTileNetworkJoiner.java

```diff
@@ -88,11 +88,46 @@ public void handlePacketData(ByteBuf dataStream) throws Exception
 	@Override
 	public ArrayList<Object> getNetworkedData(ArrayList<Object> data)
 	{
+		ArrayList<Object> childData = new ArrayList<>();
 		for(byte slotValue : tileSideMap.keySet())
 		{
-			tileSideMap.get(slotValue).getNetworkedData(data);
+			tileSideMap.get(slotValue).getNetworkedData(childData);
+			data.addAll(childData);
+			childData.clear();
 		}
 		
 		return data;
 	}
+
+	/**
+	 * Determines whether or not an {@link ITileNetwork} joiner is needed and, if so, adds a single byte header
+	 * used to route packets inside multipart containers.
+	 * @param entity The entity for which <code>getNetworkedData</code> is being called
+	 * @param data The network data list
+	 * @param facing The side this part is attached to or <code>null</code> for the center slot
+	 */
+	public static void addMultipartHeader(TileEntity entity, ArrayList<Object> data, EnumFacing facing)
+	{
+		int tileNetworkParts = 0;
+		IMultipartContainer container = MultipartMekanism.getContainer(entity.getWorld(), entity.getPos());
+		if(container != null)
+		{
+			for(IPartSlot slot : container.getParts().keySet())
```

Comment:
The joiner only gets created if 2 or more parts implement `ITileNetwork` so we do need to count in order to determine if the header is actually necessary, otherwise the header doesn't always get removed properly.